Resources/doc/reference/configuration/user-context.rst

```diff
@@ -120,6 +120,16 @@ request. However, when you decide to cache hash responses, you must invalidate
 them when the user context changes, particularly when the user logs in or out.
 This bundle provides a logout handler that takes care of this for you.
 
+.. _always_vary_on_context_hash:
+
+``always_vary_on_context_hash``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**type**: ``boolean`` **default**: ``true``
+
+Set ``always_vary_on_context_hash`` to  ``false`` to disable always adding X-User-Context-Hash or Cookie, Authorization in Vary header.
```

Comment:
ah, i think i got it now. for the hash lookup we handle that in onKernelRequest. but onKernelResponse we still add the vary in case the hash header was not present, as a safety precaution against mistakes. the no vary header switch turns of the safety, but not the hash lookup vary.

okay, i agree. can you reformulate this to be more clear? it does not automatically add the vary on X-User-Hash and switches of the safety precaution of adding vary on the user identifier headers if no hash was in the request.